src/gluonts/dataset/util.py

```diff
@@ -87,20 +86,33 @@ def _split(
     return left, right
 
 
-def _list_files(directory: Path) -> Iterator[Path]:
-    for dirname, _, filenames in os.walk(directory):
-        for filename in filenames:
-            yield Path(dirname, filename)
+def _list_files(paths: List[Path]) -> List[Path]:
+    files = []
+    for p in paths:
+        if p.exists() and p.is_file():
```

Comment:
Shouldn't just `is_file()` work here, since if it doesn't exist, it's also a file.

Reply:
Also, I think this can be more easily done with `Path.rglob`.

Will look into this later or tomorrow.